invokeai/app/services/model_install/model_install_default.py

```diff
@@ -0,0 +1,405 @@
+
+from invokeai.app.services.config import InvokeAIAppConfig
+from invokeai.app.services.events import EventServiceBase
+from invokeai.app.services.model_records import DuplicateModelException, ModelRecordServiceBase, UnknownModelException
+from invokeai.backend.model_manager.config import (
+    AnyModelConfig,
+    BaseModelType,
+    InvalidModelConfigException,
+    ModelType,
+)
+from invokeai.backend.model_manager.hash import FastModelHash
+from invokeai.backend.model_manager.probe import ModelProbe
+from invokeai.backend.model_manager.search import ModelSearch
+from invokeai.backend.util import Chdir, InvokeAILogger
+
+from .model_install_base import (
+    InstallStatus,
+    LocalModelSource,
+    ModelInstallJob,
+    ModelInstallServiceBase,
+    ModelSource,
+    UnknownInstallJobException,
+)
+
+# marker that the queue is done and that thread should exit
+STOP_JOB = ModelInstallJob(
+    source=LocalModelSource(path="stop"),
+    local_path=Path("/dev/null"),
+)
+
+
+class ModelInstallService(ModelInstallServiceBase):
+    """class for InvokeAI model installation."""
+
+    _app_config: InvokeAIAppConfig
+    _record_store: ModelRecordServiceBase
+    _event_bus: Optional[EventServiceBase] = None
+    _install_queue: Queue[ModelInstallJob]
+    _install_jobs: Dict[ModelSource, ModelInstallJob]
+    _logger: Logger
+    _cached_model_paths: Set[Path]
+    _models_installed: Set[str]
+
+    def __init__(
+        self,
+        app_config: InvokeAIAppConfig,
+        record_store: ModelRecordServiceBase,
+        event_bus: Optional[EventServiceBase] = None,
+    ):
+        """
+        Initialize the installer object.
+
+        :param app_config: InvokeAIAppConfig object
+        :param record_store: Previously-opened ModelRecordService database
+        :param event_bus: Optional EventService object
+        """
+        self._app_config = app_config
+        self._record_store = record_store
+        self._event_bus = event_bus
+        self._logger = InvokeAILogger.get_logger(name=self.__class__.__name__)
+        self._install_jobs = {}
+        self._install_queue = Queue()
+        self._cached_model_paths = set()
+        self._models_installed = set()
+        self._start_installer_thread()
+
+    @property
+    def app_config(self) -> InvokeAIAppConfig:  # noqa D102
+        return self._app_config
+
+    @property
+    def record_store(self) -> ModelRecordServiceBase:  # noqa D102
+        return self._record_store
+
+    @property
+    def event_bus(self) -> Optional[EventServiceBase]:  # noqa D102
+        return self._event_bus
+
+    def _start_installer_thread(self) -> None:
+        threading.Thread(target=self._install_next_item, daemon=True).start()
+
+    def _install_next_item(self) -> None:
+        done = False
+        while not done:
+            job = self._install_queue.get()
+            if job == STOP_JOB:
+                done = True
+                continue
```

Comment:
The case in which you create an installer temporarily, use it, and then want to get rid of it. Deleting the object won't deactivate its thread. A realistic use case is migrating models from one InvokeAI root directory to another. I would want to create an installer with an app_config corresponding to the new root directory so that I could install into it.

Reply:
I've renamed `release` to `__del__` so that it is only executed at garbage collection time.